realm/src/main/java/io/realm/internal/Context.java

```diff
@@ -55,6 +60,12 @@ public void executeDelayedDisposal() {
                 TableQuery.nativeClose(nativePointer);
             }
             abandonedQueries.clear();
+
+            NativeObjectReference reference;
+            while ((reference = (NativeObjectReference) rowReferenceQueue.poll()) != null) {
```

Comment:
It would be really nice if we could move this to some kind of worker thread with a BlockingQueue as this seems kinda dangerous. It would also remove the ton of context.executeDelayedDisposal() that litters the current code, but I assume that is not possible as RowAccessors has to be closed on the thread they where created?
And calling dispose as often as possible help keep this list small?